pkg/storage/chunk/objectclient/client.go

```diff
@@ -22,14 +24,39 @@ var Base64Encoder = func(key string) string {
 	return base64.StdEncoding.EncodeToString([]byte(key))
 }
 
+// TenantBase64Encoder is a variation of Base64Encoder that encodes tenant
+// and remainder of the key separately and returns the results joined by "/"
+var TenantBase64Encoder = func(key string) string {
+	data := []byte(key)
+	if i := bytes.LastIndex(data, []byte("/")); i > 0 {
+		return strings.Join([]string{
+			base64.URLEncoding.EncodeToString(data[:i]),
+			base64.URLEncoding.EncodeToString(data[i+1:]),
+		}, "/")
+	}
+	return base64.URLEncoding.EncodeToString(data)
+}
+
 // Client is used to store chunks in object store backends
 type Client struct {
 	store      chunk.ObjectClient
 	keyEncoder KeyEncoder
 }
 
 // NewClient wraps the provided ObjectClient with a chunk.Client implementation
-func NewClient(store chunk.ObjectClient, encoder KeyEncoder) *Client {
+func NewClient(store chunk.ObjectClient) *Client {
+	var encoder KeyEncoder = nil
```

Comment:
This will help pass the linter :)
```suggestion
	var encoder KeyEncoder
```